.github/actions/setup-haskell/action.yaml

```diff
@@ -19,18 +25,25 @@ runs:
       id: setup-haskell
       with:
         ghc-version: ${{ inputs.ghc-version }}
+        cabal-version: ${{ inputs.cabal-version }}
+
+    - name: Verify Haskell setup
+      shell: bash
+      run: |
+        ghc --version
+        cabal --version
 
     # Based on the official recipe for Cabal caching:
-    # https://github.com/actions/cache/blob/5a3ec84eff668545956fd18022155c47e93e2684/examples.md#haskell---cabal
+    # https://github.com/actions/cache/blob/v4.2.3/examples.md#haskell---cabal
     - name: Cache Cabal dependencies
       uses: actions/cache@v4
       with:
         path: |
           ${{ steps.setup-haskell.outputs.cabal-store }}
-          waspc/dist-newstyle
         # Caches are immutable, so we use the official recipe for simulating an updatable cache:
-        # https://github.com/actions/cache/blob/5a3ec84eff668545956fd18022155c47e93e2684/tips-and-workarounds.md#update-a-cache
-        key: ${{ runner.os }}-cabal-${{ inputs.ghc-version }}-${{ hashFiles('waspc/*.cabal', 'waspc/*.project', 'waspc/*.project.freeze') }}-${{ github.run_id }}
+        # https://github.com/actions/cache/blob/v4.2.3/tips-and-workarounds.md#update-a-cache
```

Comment:
I remember we used to have a pretty rich comment here before, but it is gone now. Would it make sense to return some of that? Because I am reading this now, and even though I went and read the link, I am not able to quickly grasp what is this about, even though it was me who wrote both this code and that comment some longer time ago. Since this is a tricky thing here, I think explaining it more in our words would be beneficial.

Reply:
You're right, we're already getting hashing the cabal files so we don't need to update the cache.